src/Libraries/RevitNodes/Elements/FaceWall.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+
+using Autodesk.Revit.DB;
+
+using DynamoServices;
+using Revit.GeometryConversion;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+using Curve = Autodesk.Revit.DB.Curve;
+
+namespace Revit.Elements
+{
+    /// <summary>
+    /// FaceWall Element
+    /// </summary>
+    [RegisterForTrace]
+    public class FaceWall : Element
+    {
+        #region Internal Properties
+
+        /// <summary>
+        /// Internal reference to the Revit Element
+        /// </summary>
+        internal Autodesk.Revit.DB.FaceWall InternalFaceWall
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalFaceWall; }
+        }
+
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// Create from an existing Revit Element
+        /// </summary>
+        /// <param name="wall"></param>
+        private FaceWall(Autodesk.Revit.DB.FaceWall wall)
+        {
+            SafeInit(() => InitFaceWall(wall));
+        }
+
+        /// <summary>
+        /// Create a new instance of WallType, deleting the original
+        /// </summary>
+        /// <param name="location"></param>
+        /// <param name="wallType"></param>
+        /// <param name="reference"></param>
+        private FaceWall(WallLocationLine location, Autodesk.Revit.DB.WallType wallType, Autodesk.Revit.DB.Reference reference)
+        {
+            SafeInit(() => InitFaceWall(location, wallType, reference));
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        /// <summary>
+        /// Initialize a Wall element
+        /// </summary>
+        /// <param name="wall"></param>
+        private void InitFaceWall(Autodesk.Revit.DB.FaceWall wall)
+        {
+            InternalSetFaceWall(wall);
+        }
+
+        /// <summary>
+        /// Initialize a Wall element
+        /// </summary>
+        /// <param name="location"></param>
+        /// <param name="wallType"></param>
+        /// <param name="reference"></param>
+        private void InitFaceWall(WallLocationLine location, Autodesk.Revit.DB.WallType wallType, Autodesk.Revit.DB.Reference reference)
+        {
+            TransactionManager.Instance.EnsureInTransaction(Document);
+
+            
+            var wallElem = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.FaceWall>(Document);
+
+            // there is no way of comparing the existing element to the new reference.
```

Comment:
If it is the issue that the old FaceWall can not be reused, then this is a limitation for element binding.